illumio_connector.py

```diff
@@ -1,249 +1,341 @@
-                ), resp_json
-            )
+    def initialize(self):
+        """Initialize the global variables with its value and validate it."""
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+        if not isinstance(self._state, dict):
+            self.debug_print("Resetting the state file with the default format")
+            self._state = {"app_version": self.get_app_json().get("app_version")}
 
-        return self._process_response(r, action_result)
+        # get the asset config
+        config = self.get_config()
+        self._api_key = config["api_key"]
+        self._api_secret = config["api_secret"]
+        self._hostname = config["hostname"]
 
-    def _handle_test_connectivity(self, param):
-        action_result = self.add_action_result(ActionResult(dict(param)))
+        self._port = self._validate_integers(self, config["port"], "port")
+        if self._port is None:
+            return self.get_status()
 
-        self.save_progress("Connecting to endpoint")
-        ret_val, _ = self._make_rest_call('/api/{0}/health'.format(self._api_version), action_result)
+        self._org_id = self._validate_integers(self, config["org_id"], "org_id")
+        if self._org_id is None:
+            return self.get_status()
 
-        if phantom.is_fail(ret_val):
-            self.save_progress("Test Connectivity Failed.")
-            return action_result.get_status()
+        return phantom.APP_SUCCESS
 
-        self.save_progress("Test Connectivity Passed")
-        return action_result.set_status(phantom.APP_SUCCESS)
+    def check_for_future_datetime(self, datetime_obj):
+        """
+        Checks the given datetime str isa future date or not
 
-    def _handle_create_virtual_service(self, param):
-        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
-        action_result = self.add_action_result(ActionResult(dict(param)))
+        :param datetime_obj: datetime object
+        :return : bool
+        """
+        return datetime_obj > datetime.now(tz=pytz.utc)
 
-        port = param['port']
-        protocol = param['protocol']
-        name = param.get('name', 'SOAR-{0}-{1}'.format(port, protocol))
+    def convert_to_iso(self, dt_str, action_result, key):
+        """
+        Converst input date to iso8601 datetime
 
-        virtual_service = VirtualService(
-            name=name,
-            service_ports=[
-                ServicePort(port=port, proto=convert_protocol(protocol))
-            ]
-        )
+        :param dt_str: datetime string
+        :param action_result: action result object
+        :param key: input parameter key
+        :return : status and datetime object
+        """
+        try:
+            date_time = parse(dt_str)
+            if not date_time.tzinfo:
```

Comment:
We should convert to UTC regardless - if someone passes a valid datetime with, e.g. +05:00 then this logic will keep that tz.

Reply:
Done